### DISCUSSION TITLE
Lesson 4: ModuleNotFoundError: No module named 'web3'

### DISCUSSION THREAD

**Discussion by CrispyButter** [smartcontractkit/full-blockchain-solidity-course-py#390] [Q&A]
Can anyone help with the below? Not sure why I am receiving this error. ![image](https://user-images.githubusercontent.com/94574606/142294885-85be5f7c-3937-4e50-9604-c4fe29caf1fc.png)

**Comment by CrispyButter** [comment]
It mentions line 3 but I think I did line 3 correct (see below)![image](https://user-images.githubusercontent.com/94574606/142295027-238cdac1-c90d-4bc1-b78b-1a0a84bf4143.png)

**Comment by cromewar** [comment]
Hello @CrispyButter you must install web3 first using pip, use [this](https://web3py.readthedocs.io/en/stable/quickstart.html#installation) as reference.

Best regards, Cromewar.

**Comment by CrispyButter** [comment]
Thanks cromewar, I did that previously and still receive the error though. I even tried pip3 install web3 and no luck. Please see below:


ModuleNotFoundError: No module named 'web3'
(venv) Trell@Kentrells-MacBook-Pro web3_py_simple_storage % pip install web3

Requirement already satisfied: web3 in ./venv/lib/python3.10/site-packages (5.24.0)
Requirement already satisfied: eth-account<0.6.0,>=0.5.6 in ./venv/lib/python3.10/site-packages (from web3) (0.5.6)
Requirement already satisfied: eth-utils<2.0.0,>=1.9.5 in ./venv/lib/python3.10/site-packages (from web3) (1.10.0)
Requirement already satisfied: lru-dict<2.0.0,>=1.1.6 in ./venv/lib/python3.10/site-packages (from web3) (1.1.7)
Requirement already satisfied: ipfshttpclient==0.8.0a2 in ./venv/lib/python3.10/site-packages (from web3) (0.8.0a2)
Requirement already satisfied: hexbytes<1.0.0,>=0.1.0 in ./venv/lib/python3.10/site-packages (from web3) (0.2.2)
Requirement already satisfied: requests<3.0.0,>=2.16.0 in ./venv/lib/python3.10/site-packages (from web3) (2.26.0)
Requirement already satisfied: jsonschema<4.0.0,>=3.2.0 in ./venv/lib/python3.10/site-packages (from web3) (3.2.0)
Requirement already satisfied: eth-typing<3.0.0,>=2.0.0 in ./venv/lib/python3.10/site-packages (from web3) (2.2.2)
Requirement already satisfied: websockets<10,>=9.1 in ./venv/lib/python3.10/site-packages (from web3) (9.1)
Requirement already satisfied: eth-hash[pycryptodome]<1.0.0,>=0.2.0 in ./venv/lib/python3.10/site-packages (from web3) (0.3.2)
Requirement already satisfied: eth-abi<3.0.0,>=2.0.0b6 in ./venv/lib/python3.10/site-packages (from web3) (2.1.1)
Requirement already satisfied: aiohttp<4,>=3.7.4.post0 in ./venv/lib/python3.10/site-packages (from web3) (3.8.1)
Requirement already satisfied: protobuf<4,>=3.10.0 in ./venv/lib/python3.10/site-packages (from web3) (3.19.1)
Requirement already satisfied: multiaddr>=0.0.7 in ./venv/lib/python3.10/site-packages (from ipfshttpclient==0.8.0a2->web3) (0.0.9)
Requirement already satisfied: frozenlist>=1.1.1 in ./venv/lib/python3.10/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (1.2.0)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in ./venv/lib/python3.10/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (4.0.1)
Requirement already satisfied: aiosignal>=1.1.2 in ./venv/lib/python3.10/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (1.2.0)
Requirement already satisfied: yarl<2.0,>=1.0 in ./venv/lib/python3.10/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (1.7.2)
Requirement already satisfied: charset-normalizer<3.0,>=2.0 in ./venv/lib/python3.10/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (2.0.7)
Requirement already satisfied: multidict<7.0,>=4.5 in ./venv/lib/python3.10/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (5.2.0)
Requirement already satisfied: attrs>=17.3.0 in ./venv/lib/python3.10/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (21.2.0)
Requirement already satisfied: typing-extensions>=3.6.5 in ./venv/lib/python3.10/site-packages (from async-timeout<5.0,>=4.0.0a3->aiohttp<4,>=3.7.4.post0->web3) (3.10.0.2)
Requirement already satisfied: parsimonious<0.9.0,>=0.8.0 in ./venv/lib/python3.10/site-packages (from eth-abi<3.0.0,>=2.0.0b6->web3) (0.8.1)
Requirement already satisfied: eth-keys!=0.3.2,<0.4.0,>=0.2.1 in ./venv/lib/python3.10/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (0.3.3)
Requirement already satisfied: rlp<3,>=1.0.0 in ./venv/lib/python3.10/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (2.0.1)
Requirement already satisfied: bitarray<1.3.0,>=1.2.1 in ./venv/lib/python3.10/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (1.2.2)
Requirement already satisfied: eth-keyfile<0.6.0,>=0.5.0 in ./venv/lib/python3.10/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (0.5.1)
Requirement already satisfied: eth-rlp<2,>=0.1.2 in ./venv/lib/python3.10/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (0.2.1)
Requirement already satisfied: pycryptodome<4,>=3.6.6 in ./venv/lib/python3.10/site-packages (from eth-hash[pycryptodome]<1.0.0,>=0.2.0->web3) (3.11.0)
Requirement already satisfied: cytoolz<1.0.0,>=0.9.0 in ./venv/lib/python3.10/site-packages (from eth-keyfile<0.6.0,>=0.5.0->eth-account<0.6.0,>=0.5.6->web3) (0.11.2)
Requirement already satisfied: toolz>=0.8.0 in ./venv/lib/python3.10/site-packages (from cytoolz<1.0.0,>=0.9.0->eth-keyfile<0.6.0,>=0.5.0->eth-account<0.6.0,>=0.5.6->web3) (0.11.2)
Requirement already satisfied: setuptools in ./venv/lib/python3.10/site-packages (from jsonschema<4.0.0,>=3.2.0->web3) (57.4.0)
Requirement already satisfied: six>=1.11.0 in ./venv/lib/python3.10/site-packages (from jsonschema<4.0.0,>=3.2.0->web3) (1.16.0)
Requirement already satisfied: pyrsistent>=0.14.0 in ./venv/lib/python3.10/site-packages (from jsonschema<4.0.0,>=3.2.0->web3) (0.18.0)
Requirement already satisfied: netaddr in ./venv/lib/python3.10/site-packages (from multiaddr>=0.0.7->ipfshttpclient==0.8.0a2->web3) (0.8.0)
Requirement already satisfied: base58 in ./venv/lib/python3.10/site-packages (from multiaddr>=0.0.7->ipfshttpclient==0.8.0a2->web3) (2.1.1)
Requirement already satisfied: varint in ./venv/lib/python3.10/site-packages (from multiaddr>=0.0.7->ipfshttpclient==0.8.0a2->web3) (1.0.2)
Requirement already satisfied: idna<4,>=2.5 in ./venv/lib/python3.10/site-packages (from requests<3.0.0,>=2.16.0->web3) (3.3)
Requirement already satisfied: certifi>=2017.4.17 in ./venv/lib/python3.10/site-packages (from requests<3.0.0,>=2.16.0->web3) (2021.10.8)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./venv/lib/python3.10/site-packages (from requests<3.0.0,>=2.16.0->web3) (1.26.7)
WARNING: You are using pip version 21.2.3; however, version 21.3.1 is available.
You should consider upgrading via the '/Users/Trell/demos/web3_py_simple_storage/venv/bin/Python3 -m pip install --upgrade pip' command.

(venv) Trell@Kentrells-MacBook-Pro web3_py_simple_storage %

**Reply by cromewar** [reply]
Seems everything is ok, which is weird because you have a problem... may be using other virtualenv?

**Comment by risktakereth** [comment]
I have the same error :/

**Reply by cromewar** [reply]
Looking again to the "error" this is actually just saying the library is already installed so I dont think is an actual error.

**Comment by cromewar** [comment]
@CrispyButter here is a possible solution: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/118#discussioncomment-1843933
